pkg/token/interactive.go

```diff
@@ -51,32 +54,54 @@ func (p *InteractiveToken) Token() (adal.Token, error) {
 	clientOpts := azcore.ClientOptions{Cloud: cloud.Configuration{
 		ActiveDirectoryAuthorityHost: authorityFromConfig.String(),
 	}}
-	cred, err := azidentity.NewInteractiveBrowserCredential(&azidentity.InteractiveBrowserCredentialOptions{
-		ClientOptions: clientOpts,
-		TenantID:      p.tenantID,
-		ClientID:      p.clientID,
-	})
-	if err != nil {
-		return emptyToken, fmt.Errorf("unable to create credential. Received: %w", err)
-	}
+	scopes := []string{p.resourceID + "/.default"}
 
-	// Use the token provider to get a new token
-	interactiveToken, err := cred.GetToken(context.Background(), policy.TokenRequestOptions{Scopes: []string{p.resourceID + "/.default"}})
-	if err != nil {
-		return emptyToken, fmt.Errorf("expected an empty error but received: %w", err)
-	}
-	if interactiveToken.Token == "" {
-		return emptyToken, errors.New("did not receive a token")
+	var token string
+	var expirationTimeUnix int64
+	var err error
+
+	if p.popClaims == nil || len(p.popClaims) == 0 {
```

Comment:
So at that part of the code, you relying on the fact that claims should not be empty if the enabled flag is passed along with the u claim right ?

Reply:
Yup, the initial validation in `options.go` ensures both flags are either passed together or not at all. Then the check when getting the token provider in `provider.go` ensures that if the pop-enabled flag is present, the pop-claims parse correctly and u-claim is present before providing pop claims to the token provider.